src/axom/core/utilities/StringUtilities.cpp

```diff
@@ -48,15 +48,20 @@ std::vector<std::string> splitLastNTokens(const std::string& input,
   auto last_pos = std::string::npos;
   auto pos = input.find_last_of(delim, last_pos - 1);
 
-  while((pos != std::string::npos) && (result.size() < n - 1))
+  if(n > 0 && !input.empty())
```

Comment:
@joshessman-llnl -- I think you added this function. Can you please confirm the new edge-case conditions.

Essentially, if the input is empty or the user asks for 0 tokens, the function now returns an empty vector (instead of a vector with an empty string, and a vector with all tokens, respectively)

Reply:
I think this is reasonable - it looks like the `result.size() < n - 1` was bogus anyways if `n == 0` since `n` is of unsigned type.

Reply:
Thanks -- it was, but had the effect of making `N` irrelevant, and returned all tokens

Reply:
As an aside, the API/signature for this function is different from the one for `split()` above.
This one returns a vector, the other has a vector as an `out` parameter.

In the future, we might want to make these consistent.

Reply:
See #624